### ISSUE TITLE
npm install failing with node-gyp build error

### ISSUE THREAD

**Issue by BrianBMN** [blinkhash/blinkhash-server#6]
Not sure if what I am seeing is a bug or build env issue. I can attach the entire debug log if needed

environment:
Node 12.16
Python 3.9
MSVS 2017
Windows 10 20H2

```
cd .\blinkhash-server
git pull
nvm use 12.16
npm install

```

gyp build error below

```
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\brian\AppData\Roaming\nvm\v12.16.0\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\brian\\AppData\\Roaming\\nvm\\v12.16.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\brian\blinkhash-server\node_modules\multi-hashing
gyp ERR! node -v v12.16.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\jest-haste-map\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules\nodemon\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! multi-hashing@git://github.com/blinkhash/blinkhash-multi-hashing.git#61611aa288f411f4ee95d9722e5ce2dce6a86e3a install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the multi-hashing@git://github.com/blinkhash/blinkhash-multi-hashing.git#61611aa288f411f4ee95d9722e5ce2dce6a86e3a install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\brian\AppData\Roaming\npm-cache\_logs\2021-05-30T02_47_32_747Z-debug.log

```

**Comment by blinkhash** [comment]
This is more an error due to the difficulties that come from building on Windows, however after noticing the issue, I was successful in doing so after about a day of trying. I'll include a guide in the documentation once the software is finished, but I'd still recommend getting Linux partitioned or on Virtualbox if at all possible.

**Comment by BrianBMN** [comment]
I was able to get it built on Linux (Ubuntu 20.04) without issue. I gave up on a Windows build after about a day. In addition to
`build-essential` I also had to install `libgmp3-dev`. I haven't tested it with a coin daemon yet.

I am not the best developer, but can certainly help test :D

**Comment by blinkhash** [comment]
Awesome, glad to hear it! Yeah I tried for a bit to get a Windows build working after seeing the issue, had to work through a lot of the C code in https://github.com/blinkhash/blinkhash-multi-hashing which was a pain, although it should be somewhat buildable now if necessary. I removed the `libgmp3-dev` dependency also with the recent updates as it was only being used for an algorithm that barely sees any use in active coins. Will let you know if I need any testing done! Feel free to join the Discord (https://discord.gg/8xtHZFKJQY) to keep track of ongoing development and any questions.
